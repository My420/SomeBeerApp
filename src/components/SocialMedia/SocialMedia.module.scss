@import '../../sass/variables.scss';

.list {
  display: flex;
  margin-top: 4px;
  list-style: none;
}

.item {
  width: 30px;
  height: 30px;
  margin: 0 5px;
}

.item-skype,
.item-github,
.item-email,
.link-skype,
.link-github,
.link-email {
  background-color: transparent;
}

.link {
  display: block;
  width: 30px;
  height: 30px;
}

.link:hover .path,
.link:focus:hover .path {
  fill: $menu-hover-color;
}

.link:focus {
  outline: none;
}

.link:focus .path {
  fill: $social-focus-color;
}

.link:active .path {
  opacity: 0.5;
  fill: $menu-hover-color;
}

@media (max-width: $MPC-max-width) {
  .item,
  .link {
    width: 26px;
    height: 26px;
  }
}

@media (max-width: $Tablet9-max-width) {
  .item,
  .link {
    width: 22px;
    height: 22px;
  }
}

@media (max-width: $Tablet7-max-width) {
  .item,
  .link {
    width: 18px;
    height: 18px;
  }
}

@media (max-width: $Mobile-max-width) {
  .item,
  .link {
    width: 14px;
    height: 14px;
  }
}
