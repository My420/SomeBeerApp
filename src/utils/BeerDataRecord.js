import { Record } from 'immutable';

const BeerDataRecord = Record(
  {
    id: null,
    name: null,
    tagline: null,
    first_brewed: null,
    description: null,
    image_url: null,
    abv: null,
    ibu: null,
    target_fg: null,
    target_og: null,
    ebc: null,
    srm: null,
    ph: null,
    attenuation_level: null,
    volume: {},
    boil_volume: {},
    method: {},
    ingredients: {},
    food_pairing: [],
    brewers_tips: null,
    contributed_by: null
  },
  'BeerDataRecord'
);

export default BeerDataRecord;
