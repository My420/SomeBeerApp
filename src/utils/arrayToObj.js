/* eslint no-param-reassign: "error" */

const arrayToObj = arr => {
  return arr.reduce((prevValue, element) => {
    prevValue[element.id] = element;
    return prevValue;
  }, {});
};

export default arrayToObj;
