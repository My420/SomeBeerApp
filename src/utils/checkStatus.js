const checkStatus = response => {
  const responseStatus = +response.status;
  if (responseStatus >= 200 && responseStatus < 300) {
    return response;
  }
  throw new Error(response.statusText);
};

export default checkStatus;
