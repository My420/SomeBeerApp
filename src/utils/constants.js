export const SHOP_NAME = 'Some Beer Shop';

export const LINK = {
  HOME: '/',
  CATALOG: '/catalog',
  ROULETTE: '/roulette',
  ABOUT: '/about',
  FAVORITE: '/favorite',
  CART: '/cart'
};

export const NAVIGATION_LINKS = [
  [LINK.HOME, 'Home'],
  [LINK.CATALOG, 'Catalog'],
  [LINK.ROULETTE, 'Roulette!'],
  [LINK.FAVORITE, 'Favorite'],
  [LINK.ABOUT, 'About'],
  [LINK.CART, 'Cart']
];

export const BROWSER_WIDTH_RATIO = {
  LPC: 4,
  MPC: 3,
  TABLET_9: 2,
  TABLET_7: 1,
  MOBILE: 0
};

export const BROWSER_MIN_WIDTH = {
  LPC: 1200,
  MPC: 993,
  TABLET_9: 768,
  TABLET_7: 480,
  MOBILE: 320
};

export const SOCIAL_LINK = {
  GITHUB: 'https://github.com/My420',
  EMAIL: 'mailto:klinovitsky.aleksey@gmail.com',
  SKYPE: 'skype:live:klinovitsky.aleksey?chat'
};

export const AUTHOR = 'Klinovitsky Aleksey';

export const POPULAR_BEER__GET_REQUEST = 'POPULAR_BEER__GET_REQUEST';
export const POPULAR_BEER__GET_SUCCESS = 'POPULAR_BEER__GET_SUCCESS';
export const POPULAR_BEER__GET_FAILURE = 'POPULAR_BEER__GET_FAILURE';

export const API_HOST = 'https://api.punkapi.com/v2/';

export const POPULAR_BEER_AMOUNT = 8;

export const LOAD_ERROR_MESSAGE =
  'Could not download data, please refresh the page';

export const ERROR_TITLE = 'Oops! Something went wrong.';
export const ERROR_ACTION = 'Please refresh page or try again later.';
