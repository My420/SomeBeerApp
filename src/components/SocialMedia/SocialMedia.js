import React from 'react';
import styles from './SocialMedia.module.scss';
import ListGroup from '../ListGroup/ListGroup';
import ListGroupItem from '../ListGroupItem/ListGroupItem';
import IconGithub from '../IconGithub/IconGithub';
import IconEmail from '../IconEmail/IconEmail';
import IconSkype from '../IconSkype/IconSkype';
import { SOCIAL_LINK } from '../../utils/constants';

const socialMedia = [
  {
    title: 'GitHub profile',
    href: SOCIAL_LINK.GITHUB,
    displayedText: <IconGithub pathClass={styles.path} />,
    uniqueModifier: 'github'
  },
  {
    title: 'Email Address',
    href: SOCIAL_LINK.EMAIL,
    displayedText: <IconEmail pathClass={styles.path} />,
    uniqueModifier: 'email'
  },
  {
    title: 'Skype',
    href: SOCIAL_LINK.SKYPE,
    displayedText: <IconSkype pathClass={styles.path} />,
    uniqueModifier: 'skype'
  }
];

const SocialMedia = () => {
  return (
    <ListGroup className={styles.list}>
      {socialMedia.map(elem => {
        const { title, href, displayedText, uniqueModifier } = elem;

        const className = {
          listGroupItem: `${styles.item} ${styles[`item-${uniqueModifier}`]}`,
          link: `${styles.link} ${styles[`link-${uniqueModifier}`]}`
        };

        return (
          <ListGroupItem
            className={className.listGroupItem}
            key={uniqueModifier}
          >
            <a className={className.link} href={href} title={title}>
              {displayedText}
            </a>
          </ListGroupItem>
        );
      })}
    </ListGroup>
  );
};

export default SocialMedia;
