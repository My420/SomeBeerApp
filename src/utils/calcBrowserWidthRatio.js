import { BROWSER_MIN_WIDTH, BROWSER_WIDTH_RATIO } from './constants';

const calcBrowserWidthRatio = width => {
  if (width > BROWSER_MIN_WIDTH.LPC) {
    return BROWSER_WIDTH_RATIO.LPC;
  }
  if (width > BROWSER_MIN_WIDTH.MPC) {
    return BROWSER_WIDTH_RATIO.MPC;
  }
  if (width > BROWSER_MIN_WIDTH.TABLET) {
    return BROWSER_WIDTH_RATIO.TABLET_9;
  }
  if (width > BROWSER_MIN_WIDTH.MOBILE) {
    return BROWSER_WIDTH_RATIO.TABLET_7;
  }
  return BROWSER_WIDTH_RATIO.MOBILE;
};

export default calcBrowserWidthRatio;
