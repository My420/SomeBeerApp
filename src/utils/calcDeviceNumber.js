import { DEVICE_MIN_WIDTH, DEVICE_NUMBER } from './constants';

const calcDeviceNumber = width => {
  if (width > DEVICE_MIN_WIDTH.LPC) {
    return DEVICE_NUMBER.LPC;
  }
  if (width > DEVICE_MIN_WIDTH.MPC) {
    return DEVICE_NUMBER.MPC;
  }
  if (width > DEVICE_MIN_WIDTH.TABLET) {
    return DEVICE_NUMBER.TABLET;
  }
  return DEVICE_NUMBER.MOBILE;
};

export default calcDeviceNumber;
