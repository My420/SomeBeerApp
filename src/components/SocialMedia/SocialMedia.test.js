import React from 'react';
import { shallow } from 'enzyme';
import SocialMedia from './SocialMedia';
import { SOCIAL_LINK } from '../../utils/constants';

describe('Test <SocialMedia /> component', () => {
  const wrapper = shallow(<SocialMedia />);

  test('should correct render component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should contain a link to my githab page', () => {
    const link = wrapper.find(`a[href='${SOCIAL_LINK.GITHUB}']`);
    expect(link).toHaveLength(1);
  });

  test('should contain a link to my email', () => {
    const link = wrapper.find(`a[href='${SOCIAL_LINK.EMAIL}']`);
    expect(link).toHaveLength(1);
  });

  test('should contain a link to my skype', () => {
    const link = wrapper.find(`a[href='${SOCIAL_LINK.SKYPE}']`);
    expect(link).toHaveLength(1);
  });
});
