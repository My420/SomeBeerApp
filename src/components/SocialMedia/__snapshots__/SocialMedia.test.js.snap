// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <SocialMedia /> component should correct render component 1`] = `
<ListGroup
  className="list"
  tag="ul"
>
  <ListGroupItem
    className="item item-github"
    key="github"
  >
    <a
      className="link link-github"
      href="https://github.com/My420"
      title="GitHub profile"
    >
      <IconGithub
        pathClass="path"
        wrapperClass="container"
      />
    </a>
  </ListGroupItem>
  <ListGroupItem
    className="item item-email"
    key="email"
  >
    <a
      className="link link-email"
      href="mailto:klinovitsky.aleksey@gmail.com"
      title="Email Address"
    >
      <IconEmail
        pathClass="path"
        wrapperClass="container"
      />
    </a>
  </ListGroupItem>
  <ListGroupItem
    className="item item-skype"
    key="skype"
  >
    <a
      className="link link-skype"
      href="skype:live:klinovitsky.aleksey?chat"
      title="Skype"
    >
      <IconSkype
        pathClass="path"
        wrapperClass="container"
      />
    </a>
  </ListGroupItem>
</ListGroup>
`;
